bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/target/env/GuideEnv.scala

```diff
@@ -0,0 +1,117 @@
+package edu.gemini.spModel.target.env
+
+import edu.gemini.spModel.guide.GuideProbe
+import edu.gemini.spModel.target.SPTarget
+
+import scalaz._
+import Scalaz._
+
+import GuideEnv.Manual
+
+/** A pair of an `AutomaticGroup` guide group and, optionally, a non-empty list
+  * of manual groups.  The `GuideEnv` maintains the concept of a "primary"
+  * group.  It is the group that will be used for guiding at night. The
+  * "primary" group is either the `AutomaticGroup` or else one of the various
+  * `ManualGroup`s (if any).  If the `manual` group option is defined and if the
+  * contained `OptsList` has a focused element, it is the "primary" group.
+  * Otherwise, the automatic group is considered to be the primary.
+  */
+final case class GuideEnv(auto: AutomaticGroup, manual: Option[OptsList[ManualGroup]]) {
```

Comment:
`OptsList` was invented for the model rework.  In keeping with the theme, it's the same concept as the Java `OptionsList` but better models the concept.  It's a `OneAnd[List, A] \/ Zipper[A]`.  So it is a non-empty list that optionally has a selected element.